lib/workbook.js

```diff
@@ -559,34 +558,49 @@ class Workbook {
                 item[prop] = vals;
             }
         }
+
+        // Cleanup pass: remove any isReverse_* properties that are empty
+        for (let item of this.crate.entities()) {
+            for (let prop of Object.keys(item)) {
+                if (prop.startsWith("isReverse_")) {
+                    const v = item[prop];
+                    const empty = v === "" || v === null || v === undefined ||
+                        (Array.isArray(v) && v.length === 0);
+                    if (empty) {
+                        this.crate.deleteProperty(item, prop);
+                    }
+                }
+            }
+        }
     }
 
     backLinkItem(item) {
-        // TODO - need to check if the back link is already there
+        // Add backlinks only when they are not already present
         for (let p1 in defaults.back_links) {
             const p2 = defaults.back_links[p1];
             if (item[p1]) {
                 for (let i of item[p1]) {
-                    if (i['@id']) {
+                    if (i && i['@id']) {
                         const target = this.crate.getEntity(i['@id']);
                         if (target) {
-                            this.crate.addValues(target, p2, item);
-                        }
-                    }
-                }
-            }
-            /* This was causing duplicates
-            if (item[p2]) {
-                for (let i of item[p2]) {
-                    if (i['@id']) {
-                        const target = this.crate.getEntity(i['@id']);
-                        if (target) {
-                            this.crate.addValues(target, p1, item);
+                            // Check whether target already has a backlink to `item`
```

Comment:
is this the best place to put this? How are blanks getting in there in the first place. would be better to NOT do another complete pass if you can fix it further upstream